unturned/OpenMod.Unturned/Zombies/Events/ZombieEventsListener.cs

```diff
@@ -66,34 +67,35 @@ private void Events_OnZombieAlertingPosition(Zombie nativeZombie, ref Vector3 po
             cancel = @event.IsCancelled;
         }
 
-        private void Events_OnZombieDamaging(Zombie nativeZombie, ref ushort damageAmount, ref Vector3 ragdoll,
-            ref ERagdollEffect ragdollEffect, ref bool trackKill,
-            ref bool dropLoot, ref EZombieStunOverride stunOverride, out bool cancel)
+        private void DamageTool_damageZombieRequested(ref DamageZombieParameters parameters, ref bool shouldAllow)
         {
-            UnturnedZombie zombie = new UnturnedZombie(nativeZombie);
+            UnturnedZombie zombie = new UnturnedZombie(parameters.zombie);
+            var player = GetUnturnedPlayer(parameters.instigator as Player);
 
             UnturnedZombieDamagingEvent @event;
+            var damageAmount = (ushort)Math.Min(65535, Math.Floor(parameters.damage * parameters.times));
 
             if (damageAmount >= zombie.Health)
             {
-                @event = new UnturnedZombieDyingEvent(zombie, damageAmount, ragdoll, ragdollEffect, trackKill, dropLoot,
-                    stunOverride);
+                @event = new UnturnedZombieDyingEvent(zombie, player, damageAmount, parameters.direction, parameters.ragdollEffect, true, true,
+                    parameters.zombieStunOverride);
             }
             else
             {
-                @event = new UnturnedZombieDamagingEvent(zombie, damageAmount, ragdoll, ragdollEffect, trackKill,
-                    dropLoot, stunOverride);
+                @event = new UnturnedZombieDamagingEvent(zombie, player, damageAmount, parameters.direction, parameters.ragdollEffect, true,
+                    true, parameters.zombieStunOverride);
             }
 
             Emit(@event);
 
-            damageAmount = @event.DamageAmount;
-            ragdoll = @event.Ragdoll;
-            ragdollEffect = @event.RagdollEffect;
-            trackKill = @event.TrackKill;
-            dropLoot = @event.DropLoot;
-            stunOverride = @event.StunOverride;
-            cancel = @event.IsCancelled;
+            parameters.damage = @event.DamageAmount;
+            parameters.direction = @event.Ragdoll;
+            parameters.ragdollEffect = @event.RagdollEffect;
+            parameters.instigator = @event.Instigator.Player;
+            // parameters.trackKill = @event.TrackKill;
+            // parameters.dropLoot = @event.DropLoot;
```

Comment:
Please uncomment as plugins might want to change this

Reply:
In parameters doesn't have 'trackKill' and 'dropLoot'.

Reply:
@IAmSilK

Reply:
These parameters no longer exist as the hook was moved from the direct function to a supported event. I didn't realize this in my original comment.